src/components/Modals/TaskModal.js

```diff
@@ -0,0 +1,66 @@
+import Modal from "react-bootstrap/Modal";
+import Button from "react-bootstrap/Button";
+import { useState } from "react";
+import taskStore from "../../stores/taskStore";
+import { observer } from "mobx-react";
+import "../../App.css";
+
+const TaskModal = () => {
+  const [show, setShow] = useState(false);
+  const toggleShow = () => setShow(!show);
+
+  const [task, setTask] = useState({
+    name: "",
+    description: "",
+    status: false,
+    priority: "low",
+  });
+
+  const handleChange = (e) => {
+    setTask({ ...task, [e.target.name]: e.target.value });
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    taskStore.addTask(task);
+  };
+
+  return (
+    <div className="Modal">
+      <Button variant="primary" onClick={toggleShow}>
+        Add New Task
+      </Button>
+
+      <Modal show={show} onHide={toggleShow} onSubmit={handleSubmit}>
```

Comment:
Wrap the modal content starting from the modal body to the end of the modal footer with a ``<form>`` tag. 
Give the form tag an ``onSubmit={handleSubmit}`` and remove it from the ``<Modal>`` tag. The user submits a form and not the whole modal.